tests/test_cliUtils.py

```diff
@@ -366,5 +368,60 @@ def test_exist(self):
                 self.assertIn('"foo.txt" should not exist.', result.output)
 
 
+class MWCommandTest(unittest.TestCase):
+    def setUp(self):
+        self.runner = click.testing.CliRunner()
+        self.ctx = None
+
+    def testCaptureOptions(self):
+        """Test that command options are captured in order."""
+
+        @click.command(cls=MWCommand)
+        @click.argument("ARG_A", required=False)
+        @click.argument("ARG_B", required=False)
+        @click.option("-o", "--an-option")
+        @click.option("-s", "--second-option")
+        @click.option("-f", is_flag=True)
+        @click.option("-d/-n", "--do/--do-not", "do_or_do_not")
+        @click.option("--multi", multiple=True)
+        @click.pass_context
+        def cmd(ctx, arg_a, arg_b, an_option, second_option, f, do_or_do_not, multi):
+            self.assertIsNotNone(ctx)
+            self.ctx = ctx
+
+        for args, expected in (
+            ("--an-option foo --second-option bar", None),
```

Comment:
I'm not sure what is going on here. Why does this one seem to expand to `None` but a few lines below the same option does turn up in the expected list when it includes an `=` sign? I think I need a bit more explanation as to what is happening.

Reply:
`None` in the second item in the tuple means that the expected value is exactly the same as the input value (see line 418). I will add a comment to make that more clear.